### PR TITLE
MLHR-1877 #resolve #comment moved DTFile implementation and Cache man…

### PR DESCRIPTION
**Pull request by otterc**
…ager to library

### CODE REVIEW

**Comment by otterc** [comment]
@tweise @siyuanh please review and merge

**Comment by otterc** [comment]
I suspect that the DTFileTest which fails was even failing before my change. Since this code was in contrib before and the tests in contrib are only manually run, this wasn't seen before.

**Comment by tweise** [comment]
@siyuanh please have a look

**Comment by otterc** [comment]
@siyuanh I have fixed the test. Can you please merge this pull request. The change just moves these files from contrib to lib. Made a few changes to CacheManager as it was using a higher version of guava which wasn't what we were using in library. Also took liberty to make few style fixes to CacheManager.

File: library/src/main/java/org/apache/hadoop/io/file/tfile/CacheManager.java [hunk 1]
```diff
@@ -110,41 +116,45 @@
...
+    return SINGLE_CACHE.getIfPresent(key);
   }
 
-  public static final void invalidateKeys(Collection<String> keys)
+  public static void invalidateKeys(Collection<String> keys)
   {
-    if (singleCache != null)
-      singleCache.invalidateAll(keys);
+    if (SINGLE_CACHE != null)
+      SINGLE_CACHE.invalidateAll(keys);
```
Comment (siyuanh) [inline library/src/main/java/org/apache/hadoop/io/file/tfile/CacheManager.java:146]: Fix the cs violation

**Pull request closed without merging**